src/parser/ParseException.h

```diff
@@ -1,33 +1,49 @@
 // Copyright 2014, University of Freiburg,
 // Chair of Algorithms and Data Structures.
-// Author: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+// Authors:
+//   2014-2017 Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+//   2022      Julian Mundhahs (mundhahj@tf.informatik.uni-freiburg.de)
 #pragma once
 
 #include <exception>
 #include <string>
+#include <string_view>
 
+// Contains Metadata (position of the faulty clause) about a ParseException that
+// occurred.
 struct ExceptionMetadata {
-  std::string query_;
-  size_t startIndex = 0;
-  size_t stopIndex = 0;
+  // Query that the exception occurred in. This is currently needed because the
+  // parse doesn't parse the whole query in one piece. It therefore can only
+  // determine the position in the current (sub-)query.
+  std::string_view query_;
+  // Start and Stop Index of the clause that cause the exception in query_.
```

Comment:
Sorry that I didn't catch this the first time:
Is the token at the stop index still part of the error, or is it off by one?
In the first case, the condition `startIndex_ < stopIndex_` is wrong,
In the other case the length of the `middle` substring is too long.
I would prefer converting those to `beginIndex_` and `endIndex`, s.t. the interval is
[begin, end) (end not inclusive), because it is more idiomatic.